signer/build.rs

```diff
@@ -0,0 +1,4 @@
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    tonic_build::compile_protos("src/network/message.proto")?;
```

Comment:
Personally I dislike checking in generated code.  It needs to be part of the build anyway and not checking it in avoid having anything stale.

What does everyone else think about this?

Reply:
I used `.generated-sources` in emily, I'd stick to that idiom unless we prefer just `.generated` at which point I can change the emily ones to match.

Reply:
I don't have an opinion on the directory name (`generated` vs `generated-sources`), but I definitely like having it in a place where my IDE can help me. The default directory, which is the `OUT_DIR` environment variable that `cargo` sets, is not good for `rust-analyzer` and whatever RustRover uses.

Placing it in the existing `.generated-sources` folder is fine too, my IDE doesn't have issues with it. But having it in the `signer` crate might be nice if we want to manually implement foreign traits on the generated types. So to clarify, it doesn't need to be checked in, just generated to the repo.

Reply:
Interesting, my editor has no problems navigating to the generated sources - and it's also using `rust-analyzer`. I would ideally like to have all the build artifacts within the `target` directory. I would love to explore your IDE setup and try to understand why it cannot access the generated code in this case.

I also often find it helpful when developing with generated code to use `cargo doc` locally to browse the documentation.

Reply:
Okay, I chatted with @netrome about this and, embarrassingly enough, it turns out that it was how I configured my IDE 😞. `rust-analyzer` will find it if enable `cargo build scripts`. I don't know why I disabled it, maybe some other issue that has since been fixed /shrug. No need for any changes here.

Reply:
Definitely nothing to be embarrassed about. Thank you for bearing with me :pray: